packages/editor-ui/src/components/ParameterInput.vue

```diff
@@ -952,6 +958,34 @@ onMounted(() => {
 	});
 });
 
+const { height } = useElementSize(wrapper);
+
+const isSingleLineInput = computed(() => {
+	if (isTextarea.value && !isModelValueExpression.value) {
+		return false;
+	}
+
+	/**
+	 * There is an awkward edge case here with text boxes that automatically
+	 * adjust their row count based on their content:
+	 *
+	 * If we move the overrideButton to the options row due to going multiline,
+	 * the text area gains more width and might return to single line.
+	 * This then causes the overrideButton to move inline, causing a loop and flickering UI.
+	 *
+	 * To avoid this, we treat 2 rows of input as single line, if we are already single line.
+	 */
+	if (isSingleLineInput.value) {
+		return height.value <= 70;
+	} else {
+		return height.value <= 35;
+	}
+});
+
+defineExpose({
```

Comment:
Since you asked for feedback on this, this looks good, nice use of `defineExpose` 😀